src/lib/format/l20n/ast/parser.js

```diff
@@ -7,18 +7,20 @@ const MAX_PLACEABLES = 100;
 
 export default {
   parse: function(env, string, pos = false) {
-    this._source = string;
-    this._index = 0;
-    this._length = string.length;
-    this._curEntryStart = 0;
-    this._config = {
-      pos: pos
-    };
-
-    if (pos !== true) {
-      AST.Node.prototype.setPosition = function() {};
-    }
-    return this.getResource(pos);
+    AST.Node._config.pos = pos;
+
+    const parseContext = this.getNewParseContext(env, string, pos);
```

Comment:
Currently, there's no `ParseContext` that could be used in a snippet that you suggested. And if I name the default exported object `ParseContext` I still cannot do `new ParseContext()` because it is an object, not a function:

```
const ParseContext = {};
const parseContext = new ParseContext()
```

So maybe I don't understand your comment.